tests/docker/test-image.sh

```diff
@@ -11,150 +11,167 @@ image_root_path="${script_path}/../../utils/docker"
+  local value="$(sed -n -e "${sed_expr}" "${filename}")"
+  echo "Extracted ${tag}: '${value}'" &>2
+  echo "${value}"
+
+  return 0
+}
+
+########## main ###########
+
+# absolute path of the current script
+script_path="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# initialize variables to default values, if any
 GALAXY_URL=''
 GALAXY_API_KEY=''
 GALAXY_NETWORK=''
-repo_url=''
-GIT_BRANCH=''
-repo_branch=''
 
-# options
-opts="$@"
+# other options to be passed to build-image.sh
+OTHER_ARGS=()
 
 # disable debug
 debug=""
 
-# parse args
-while test $# -gt 0
-do
-    case "$1" in
-        -h|--help)
-            print_usage
-            exit 0
-            ;;
-        --server )
-            [ $# -ge 2 ] || { echo "Missing value for '$1'"; exit 1; }
-            GALAXY_URL="$2"
-            shift
-            ;;
-        --api-key)
-            [ $# -ge 2 ] || { echo "Missing value for '$1'"; exit 1; }
-            GALAXY_API_KEY="$2"
-            shift
-            ;;
-        --network )
-            [ $# -ge 2 ] || { echo "Missing value for '$1'"; exit 1; }
-            GALAXY_NETWORK="$2"
-            shift
-            ;;
-        --url|--repo-url)
-            [ $# -ge 2 ] || { echo "Missing value for '$1'"; exit 1; }
-            repo_url="--url $2"
-            shift
-            ;;
-        --branch|--repo-branch)
-            [ $# -ge 2 ] || { echo "Missing value for '$1'"; exit 1; }
-            GIT_BRANCH=$2
-            repo_branch="--branch $2"
-            shift
-            ;;
-        --debug )
-            debug="--debug"
-            shift
-            ;;
-        --*)
-            print_usage
-            exit -1
-            ;;
-        *)
-            # support only the first argument; skip all remaining
-            if [[ -z ${image_type} ]]; then
-                image_type=${1}
-            fi
-            ;;
-    esac
-    shift
-done
-
-# check required options
-settings=(image_type GALAXY_URL GALAXY_API_KEY)
-for s in ${settings[@]}; do
-    if [[ -z ${!s} ]]; then
-        echo "No ${s} provided." >&2
-        exit 99
-    fi
-done
-
-# check image type
-if [[ -z ${image_type} ]]; then
-    image_type="minimal"
-    echo "No image type provided. Using the default : ${image_type}">&2
-elif [[ ${image_type} != "minimal" && ${image_type} != "develop" ]]; then
-    echo -e "\nERROR: '${image_type}' not supported! Use 'minimal' or 'develop'."
-    exit 99
-else
-    export IMAGE_TYPE="${image_type}"
-fi
+parse_args "${@}"
+
+# copy the wft4galaxy script
+docker_runner="${script_path}/../../wft4galaxy/app/docker_runner.py"
+
+echo "Building docker image" >&2
+build_image="${script_path}/../../utils/docker/build-image.sh"
 
-# extract git info & Docker image name
-source ${image_root_path}/set-git-repo-info.sh ${repo_url} ${repo_branch}
-source ${image_root_path}/set-docker-image-info.sh
+# Use a temporary file to capture the name of the image being created
+tmpfile=$(mktemp --tmpdir wft4galaxy-test-image.XXXXX)
```

Comment:
Option ``--tmpdir`` is not supported on macOS.
Use the following workaround which produces the same result on both Linux and macOS:
```
tmpfile=$(mktemp --tmpdir wft4galaxy-test-image.XXXXX 2>/dev/null ||  mktemp -t  wft4galaxy-test-image.XXXXX)
```